src/Monotonic.cpp

```diff
@@ -209,10 +364,19 @@ class MonotonicVisitor : public IRVisitor {
 
     void visit_lt(const Expr &a, const Expr &b) {
         a.accept(this);
-        Monotonic ra = result;
+        ConstantInterval ra = result;
         b.accept(this);
-        Monotonic rb = result;
-        result = unify(flip(ra), rb);
+        ConstantInterval rb = result;
+        result = unify(negate(ra), rb);
+        // If the result is bounded, limit it to [-1, 1]. The largest
```

Comment:
Is the result limited to this interval? Could `result.min` be less than `-1`, or `result.max` be greater than `1`, which would produce an interval not limited to this range?

Reply:
the result isn't limited to this, but this logic then does limit it explicitly below.

Reply:
I'm not convinced, what if `result = [-10, -5]` here, then after the following casework, it would be `[-1, -5]`, which is an incorrect interval.